openfe/setup/molecule.py

```diff
@@ -42,3 +79,26 @@ def __hash__(self):
 
     def __eq__(self, other):
         return hash(self) == hash(other)
+
+    def to_sdf(self) -> str:
+        # https://sourceforge.net/p/rdkit/mailman/message/27518272/
+        mol = self.rdkit
+        sdf = [Chem.MolToMolBlock(mol)]
+        for prop in mol.GetPropNames():
+            # always output as this version of OpenFE
+            if prop == "ofe-version":
+                val = openfe.__version__
+            else:
+                val = mol.GetProp(prop)
+            sdf.append('>  <%s>\n%s\n' % (prop, val))
+        sdf.append('$$$$\n')
+        return "\n".join(sdf)
+
+    @classmethod
+    def from_sdf_string(cls, sdf_str: str):
+        # https://sourceforge.net/p/rdkit/mailman/message/27518272/
+        supp = Chem.SDMolSupplier()
+        supp.SetData(sdf_str)
+        mols = list(supp)
```

Comment:
`list()` isn't wise on things (sdf files) which can sometimes be Gb.  I'd sooner just call `next()` on it then maybe check that the iterator is spent?